### ISSUE TITLE
Pass matrix from excel to r

### ISSUE THREAD

**Issue by glitests** [sdllc/Basic-Excel-R-Toolkit#13]
I wanted to see what the form of the matrix would be when passed from Excel to R, with this simple code.
```
createDF<- function(mat){
  data <- as.data.frame(mat, ncol=3)
  return(data)
}

```
![image](https://cloud.githubusercontent.com/assets/18304035/14668254/f5f48dba-0715-11e6-9698-01fc6314f44e.png)

It seems to return 0's when the selected matrix is a mixture of types. What am I doing wrong?

**Comment by duncanwerner** [comment]
Well there are a bunch of things going on here (one of which is a bug).  Matrices in R do not have mixed types.  Matrices are vectors (arrays, roughly) of a single type.  

If you pass in an Excel range that has mixed types, we give you a list of lists, which preserves type.  This is more complicated to deal with, but we believe preserving type is more useful than having a matrix structure.

A data frame is also a list of lists, and should be able to handle what you are doing.  It seems like we are not converting df -> excel properly here.  I'll check what is happening.  For now, you could convert to a matrix when you return it

```
return( matrix(unlist(data),nrow=nrow(data)));
```

but as noted above that will convert everything to strings which is not what you want (it will work, though).

**Comment by duncanwerner** [comment]
Actually now that I think about it, data frames are lists, the members of which are either lists or vectors.  In your case they are lists, which we're not handling properly.  if you return a data frame that contains vectors, this would work.  Still, we should handle those lists properly.

```
# list of vectors
test <- function(){ 
	df <- data.frame( A=c("a", "b"), B=c( 1, 2 ), stringsAsFactors=F );
	print( str( df ));
	df;
}
```

**Comment by duncanwerner** [comment]
Fixed in 5aa9dfd98e078332c52208756c444b38df4a2ad1 (pending release).

**Comment by duncanwerner** [comment]
Should be resolved in release v0.70 (along with other type translation issues).
